osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CreateServiceFunctionChainTaskTest.java

```diff
@@ -0,0 +1,171 @@
+	public ExpectedException exception = ExpectedException.none();
+
+	@Mock
+	private EntityManager em;
+
+	@Mock
+	private EntityTransaction tx;
+
+	@Mock(answer = Answers.CALLS_REAL_METHODS)
+	private TestTransactionControl txControl;
+
+	@Mock
+	private DBConnectionManager dbMgr;
+
+	@Mock
+	private TransactionalBroadcastUtil txBroadcastUtil;
+
+	@Mock
+	private ApiFactoryService apiFactoryServiceMock;
+
+	@Mock
+	private SdnRedirectionApi sdnApi;
+
+	@InjectMocks
+	private CreateServiceFunctionChainTask task;
+
+	@Before
+	public void testInitialize() throws VmidcException {
+		MockitoAnnotations.initMocks(this);
+		this.em = InMemDB.getEntityManagerFactory().createEntityManager();
+
+		this.txControl.setEntityManager(this.em);
+		Mockito.when(this.dbMgr.getTransactionalEntityManager()).thenReturn(this.em);
+		Mockito.when(this.dbMgr.getTransactionControl()).thenReturn(this.txControl);
+
+		populateDatabase();
+	}
+
+	@AfterClass
+	public static void testTearDowm() {
+		InMemDB.shutdown();
+	}
+
+	@Test
+	public void testExecute_WithSecurityGroupSFCBinded_ExpectCreate() throws Exception {
+		// Arrange.
+		CreateServiceFunctionChainTask task = this.task.create(
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getServiceFunctionChain(),
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE);
+		registerNetworkElement(SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE, new NetworkElementImpl("NETWORK_ELEMENT_ID_2"));
+
+		// Act.
+		String taskName = task.getName();
+
+		// Assert
+		assertEquals(SecurityGroupEntityMgr.findById(this.em, SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getId())
+				.getNetworkElementId(), "NETWORK_ELEMENT_ID_2");
+		assertEquals(String.format("Creating Service Function Chain '%s' for Security Group '%s' under Project '%s'",
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getServiceFunctionChain().getName(),
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getName(),
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getProjectName()), taskName);
+	}
+
+	@Test
+	public void testExecute_WithSecurityGroupSFCBinded_ExpectUpdate() throws Exception {
+		// Arrange.
+		CreateServiceFunctionChainTask task = this.task.create(
+				NEW_SECURITY_GROUP_SAME_SFC_BINDED_UPDATE_ELEMENT_ID.getServiceFunctionChain(),
+				NEW_SECURITY_GROUP_SAME_SFC_BINDED_UPDATE_ELEMENT_ID);
+		registerNetworkElement(NEW_SECURITY_GROUP_SAME_SFC_BINDED_UPDATE_ELEMENT_ID,
+				new NetworkElementImpl("NETWORK_ELEMENT_ID_1"));
+
+		// Act.
+		task.execute();
+
+		// Assert
+		assertEquals(SecurityGroupEntityMgr
+				.findById(this.em, NEW_SECURITY_GROUP_SAME_SFC_BINDED_UPDATE_ELEMENT_ID.getId()).getNetworkElementId(),
+				"NETWORK_ELEMENT_ID_1");
+	}
+
+	@Test
+	public void testExecute_WhenSDNReturnsNullNetworkElement_ExpectException() throws Exception {
+		// Arrange.
+		CreateServiceFunctionChainTask task = this.task.create(SECURITY_GROUP_SFC.getServiceFunctionChain(),
+				SECURITY_GROUP_SFC);
+		registerNetworkElement(SECURITY_GROUP_SFC, null);
+
+		this.exception.expect(Exception.class);
```

Comment:
Is it really `Exception` type? Should be something more specific. Also in the test name should be _Throws*ExceptionType*